scripts/us_census/pep/us_pep_sex/process.py

```diff
@@ -490,26 +642,148 @@ def _state_2021(file_path: str) -> pd.DataFrame:
+            'Count_Person_Male', 'Count_Person_Female'
+        ]
+        df = pd.read_excel(file_path, skiprows=5, skipfooter=7, header=None)
+        df.columns = column_name
+        # extract geoid from file path
+        geoid = file_path[-7:-5]
+        if geoid == "0.":
+            geoid = "01"
+        df = df.query('Age == "Total"')
+        df.insert(1, 'geo_ID', 'geoId/' + geoid)
+        df = df.drop(columns=[
+            'Age', 'April2020Total', 'April2020Male', 'April2020Female',
+            'July2020Total', 'July2020Male', 'July2020Female', '2021Total'
+        ])
+        df['Year'] = '2021'
+        df['Measurement_Method'] = 'dcAggregate/CensusPEPSurvey_PartialAggregate'
+        return df
+    except Exception as e:
+        logging.fatal(f"Error processing the file {file_path}: {e}")
+
+
+def _state_2022(file_path: str) -> pd.DataFrame:
+    """
+    Process and cleans the file for state 2022.
+
+    Args:
+        file_path (str) : input file path.
+
+    Returns:
+        df (pd.DataFrame) : cleaned dataframe.
+    """
+    try:
+        column_name = [
+            'Age', 'April2020Total', 'April2020Male', 'April2020Female',
+            'July2020Total', 'July2020Male', 'July2020Female', 'July2021Total',
+            'July2021Male', 'July2021Female', '2022Total', 'Count_Person_Male',
+            'Count_Person_Female'
+        ]
+        df = pd.read_excel(file_path, skiprows=5, skipfooter=7, header=None)
+        df.columns = column_name
+        # extract geoid from file path
+        geoid = file_path[-7:-5]
+        if geoid == "0.":
+            geoid = "01"
+        df = df.query('Age == "Total"')
+        df.insert(1, 'geo_ID', 'geoId/' + geoid)
+        df = df.drop(columns=[
+            'Age', 'April2020Total', 'April2020Male', 'April2020Female',
+            'July2020Total', 'July2020Male', 'July2020Female', 'July2021Total',
+            'July2021Male', 'July2021Female', '2022Total'
+        ])
+        df['Year'] = '2022'
+        df['Measurement_Method'] = 'dcAggregate/CensusPEPSurvey_PartialAggregate'
+        return df
+    except Exception as e:
+        logging.fatal(f"Error processing the file {file_path}: {e}")
+
+
+def _state_2023(file_path: str) -> pd.DataFrame:
+    """
+    Process and cleans the file for state 2023.
+
+    Args:
+        file_path (str) : input file path.
+
+    Returns:
+        df (pd.DataFrame) : cleaned dataframe.
+    """
+    try:
+        column_name = [
+            'Age',
+            'April2020Total',
+            'April2020Male',
+            'April2020Female',
+            'July2020Total',
+            'July2020Male',
+            'July2020Female',
+            'July2021Total',
+            'July2021Male',
+            'July2021Female',
+            'July2022Total',
+            'July2022Male',
+            'July2022Female',
+            'July2023Male',
+            'July2023Female',
+            '2023Total',
```

Comment:
Can we generalize this to 2024 and future years?

Reply:
It has been generalized for future years

Reply:
Pls ignore this. For some reason, git was showing older files.